### PR TITLE
Institutions page - redux

### PR DESCRIPTION
**Pull request by wpears**
Will fail tests, beware. Putting up this PR as a place for comments/discussion on the redux implementation so it isn't just living in my branch.

Structure is fairly idiomatic redux, but please do ask questions if you have them.

The groupings on the homepage were omitted here for simplicity of implementation, but will be added back in short order.

### CODE REVIEW

**Comment by awolfe76** [comment]
@wpears - is this one ready for full review? _(There are some conflicts too.)_

**Comment by wpears** [comment]
Yep, ready to go; I have another not quite baked PR with a large percentage of the redux changes coming in next.

File: src/js/InstitutionContainer.jsx [hunk 1]
```diff
@@ -7,6 +7,7 @@ var InstitutionStatus = require('./InstitutionStatus.jsx');
 var InstitutionContainer = React.createClass({
 
   getInitialState: function(){
+    console.log('getting initial');
```
Comment (awolfe76) [inline src/js/InstitutionContainer.jsx:10]: Can you remove the logging? If you need it for the next PRs that's ok, this can just be a reminder then.

Comment (awolfe76) [inline src/js/InstitutionContainer.jsx:10]: Actually, can this whole file be removed?

File: state_shape.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
+{
```
Comment (awolfe76) [inline state_shape.json:1]: If this is really for documentation purposes, maybe a `.md` with some comments?

File: src/js/reducers/index.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+export const filings = (state = [], action) => {
+  switch (action.type) {
+  case RECEIVE_INSTITUTION:
+    return [
+      ...state,
+      ...action.institution.filings
+    ]
+  case CLEAR_FILINGS:
```
Comment (awolfe76) [inline src/js/reducers/index.js:29]: Maybe some comments around why we are returning the new state in these ways? Why its set to return what its returning? Might be helpful later to know why.

File: src/js/actions/index.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+export function clearFilings() {
+  return {
+    type: types.CLEAR_FILINGS
+  }
+}
+
+export function fetchInstitutions() {
+  return dispatch => {
```
Comment (awolfe76) [inline src/js/actions/index.js:39]: I think some comments in some of these actions would be helpful.

**Comment by awolfe76** [comment]
Overall 👍 . I think comments in some places, maybe more than I put line comments on, would be helpful. Had been thinking that for a while. Maybe as part of this redux/es6 refactor we should add some comments.

**Comment by awolfe76** [comment]
@wpears - if you think you can get https://github.com/cfpb/hmda-platform-ui/issues/187 into this PR it would be good too. That should get us to a point where we can see the home page listing the institutions.

File: src/js/InstitutionContainer.jsx (hunk 1 above)
Comment (wpears) [inline src/js/InstitutionContainer.jsx:10]: This file is removed in the next PR

File: state_shape.json (hunk 2 above)
Comment (wpears) [inline state_shape.json:1]: Yeah, this needs to be updated in general, maybe a separate PR?

**Comment by wpears** [comment]
I'll do #187 in a separate PR and comments/state shape updates in a yet another. Sound okay?

**Comment by awolfe76** [comment]
I updated #180 to include adding comments.

**Pull request merged**